deployment/terraform/create.go

```diff
@@ -340,9 +341,167 @@ func (t *Terraform) PostProcessDatabase(extAgent *ssh.ExtAgent) error {
 		}
 	}
 
+	needsReboot, err := t.HasPendingRebootDBParams()
+	if err != nil {
+		return fmt.Errorf("failed to check whether the DB has pending-reboot parameters")
+	}
+
+	if needsReboot {
+		return t.RebootDBInstances(extAgent)
+	}
+
 	return nil
 }
 
+func (t *Terraform) HasPendingRebootDBParams() (bool, error) {
+	// Build the RDS client
+	cfg, err := t.GetAWSConfig()
+	if err != nil {
+		return false, fmt.Errorf("failed to get AWS config: %w", err)
+	}
+	rdsClient := rds.NewFromConfig(cfg)
+
+	// Check in parallel whether each DB instance needs to be rebooted
+	errChan := make(chan error, len(t.output.DBCluster.Instances))
+	needsRebootChan := make(chan bool, len(t.output.DBCluster.Instances))
+	var wg sync.WaitGroup
+	for _, instance := range t.output.DBCluster.Instances {
+		wg.Add(1)
+		go func(dbId string) {
+			defer wg.Done()
+			needsReboot, err := hasPendingRebootDBParams(rdsClient, dbId)
+			needsRebootChan <- needsReboot
+			errChan <- err
```

Comment:
Is there a reason you're returning everything, instead of checking here and only returning the positives? ("to get to use the `||` below" is a valid answer)

Reply:
That's a good question. The actual answer is "because I didn't think about it much". But now that I'm merging both channels into one with a struct containing both fields, I'll keep returning both values in all instances for consistency.